File: Kattis/bits.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    int T; cin>>T;
    while(T--) {
        string X; cin>>X;
        int ans = 0;
        for(int i=0; i<X.length(); i++) {
            string s = bitset<32>(stoi(X.substr(0,i+1))).to_string();
            int temp = 0;
            for(char c:s)
                temp += (c == '1');
            ans = max(ans, temp);
        }
        cout<<ans<<endl;
    }
}

File: Kattis/bits.py
T = int(input())

def count_bits(X):
    ret = 0
    while X > 1:
        if X%2:
            ret += 1
        X //= 2
    if X%2:
        ret += 1
    return ret

for _ in range(0,T):
    X = input()
    
    ans = 0
    for i in range(0,len(X)):
        ans = max(ans, count_bits(int(X[0:i+1])))

    print(ans)
